payjoin/src/core/send/v2/mod.rs

```diff
@@ -342,18 +340,20 @@ impl Sender<WithReplyKey> {
             reply_key: post_ctx.reply_key,
         };
         MaybeFatalTransition::success(
-            SessionEvent::PollingForProposal(polling_for_proposal.clone()),
+            SessionEvent::PollingForProposal(),
             Sender { state: polling_for_proposal },
         )
     }
 
     /// The endpoint in the Payjoin URI
     pub fn endpoint(&self) -> Url { self.pj_param.endpoint().clone() }
 
-    pub(crate) fn apply_polling_for_proposal(
-        self,
-        polling_for_proposal: PollingForProposal,
-    ) -> SendSession {
+    pub(crate) fn apply_polling_for_proposal(self) -> SendSession {
+        let polling_for_proposal = PollingForProposal {
+            pj_param: self.state.pj_param,
+            psbt_ctx: self.state.psbt_ctx,
+            reply_key: self.state.reply_key,
+        };
         SendSession::PollingForProposal(Sender { state: polling_for_proposal })
```

Comment:
I probably wouldn't bother with an assignment, especially not one assigned the name `state` to elide the colon.

Reply:
Fair enough. I went back and forth but landed on this for readibility. Will remove the assignment

Reply:
Fixed